### PR TITLE
Feat/get health ratio level per user

### PR DESCRIPTION
**Pull request by iamnovichek**
1. get health ratio level per user endpoint added

### CODE REVIEW

File: data_handler/db/crud.py [hunk 1]
```diff
@@ -93,6 +93,38 @@ def delete_object(self, model: Type[Base] = None, obj_id: uuid = None) -> None:
         finally:
             db.close()
 
+    def _get_subquery(self):
```
Comment (djeck1432) [inline data_handler/db/crud.py:96]: add, please, return type

File: data_handler/db/crud.py [hunk 2]
```diff
@@ -93,6 +93,38 @@ def delete_object(self, model: Type[Base] = None, obj_id: uuid = None) -> None:
...
+        return (
+            session.query(
+                LoanState.user,
+                func.max(LoanState.block).label('latest_block')
+            )
+            .group_by(LoanState.user)
+            .subquery()
+        )
+
+    def get_latest_block_loans(self):
```
Comment (djeck1432) [inline data_handler/db/crud.py:110]: Add docstring and return type

File: data_handler/db/crud.py [hunk 3]
```diff
@@ -286,3 +318,18 @@ def get_last_interest_rate_record_by_protocol_id(
             )
         finally:
             db.close()
+
+    def get_all_block_records(self, model: Type[ModelType] = None):
```
Comment (djeck1432) [inline data_handler/db/crud.py:322]: add please return type

File: data_handler/handlers/health_ratio_level/health_ratio_handlers.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+    def __init__(self):
+        super().__init__(state_class=ZkLendState, loan_entity_class=ZkLendLoanEntity)
+
+    def calculate_health_ratio(self) -> defaultdict:
+        """
+        Calculates health ratio based on provided data.
+        :return: A dictionary of the ready health ratio data.
+        """
+        data, interest_rate_models = self.fetch_data(protocol_name=ProtocolIDs.ZKLEND.value)
```
Comment (djeck1432) [inline data_handler/handlers/health_ratio_level/health_ratio_handlers.py:100]: Split this method into 2 methods:
1. method to initialize `loan_entity`
2. Second calculate and get result

File: data_handler/main.py [hunk 5]
```diff
@@ -98,3 +98,40 @@ def get_last_interest_rate_by_block(
...
+    if protocol is None:
+        raise HTTPException(status_code=400, detail="Protocol ID is required")
+
+    if protocol not in ProtocolIDs.choices():
+        raise HTTPException(status_code=400, detail="Invalid protocol ID")
+
+    if not user_id:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="User ID is required")
+
+    row = db.query(HealthRatioLevel).filter(
```
Comment (djeck1432) [inline data_handler/main.py:128]: It should be last one `ordered_by` timestamp

File: data_handler/migrations/versions/d2fa8201b04a_add_loanstate_model.py [hunk 6]
```diff
@@ -22,28 +22,29 @@
...
-        sa.Column("deposit", sa.JSON(), nullable=True),
-        sa.Column("id", sa.UUID(), nullable=False),
-        sa.PrimaryKeyConstraint("id"),
-    )
-    op.create_index(op.f("ix_loan_state_block"), "loan_state", ["block"], unique=False)
-    op.create_index(
-        op.f("ix_loan_state_timestamp"), "loan_state", ["timestamp"], unique=False
-    )
-    op.create_index(op.f("ix_loan_state_user"), "loan_state", ["user"], unique=False)
+    # op.create_table(
```
Comment (djeck1432) [inline data_handler/migrations/versions/d2fa8201b04a_add_loanstate_model.py:27]: Why is it commented?

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: data_handler/migrations/versions/d2fa8201b04a_add_loanstate_model.py (hunk 6 above)
Comment (iamnovichek) [reply data_handler/migrations/versions/d2fa8201b04a_add_loanstate_model.py:27]: That's alembic who commented it out

**Pull request closed without merging**